mifosng-provider/src/main/java/org/mifosplatform/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java

```diff
@@ -1511,7 +1511,7 @@ public LoanTermVariationsData mapRow(final ResultSet rs, @SuppressWarnings("unus
                 .append(" where ((ls.fee_charges_amount <> if(ls.accrual_fee_charges_derived is null,0, ls.accrual_fee_charges_derived))")
                 .append(" or ( ls.penalty_charges_amount <> if(ls.accrual_penalty_charges_derived is null,0,ls.accrual_penalty_charges_derived))")
                 .append(" or ( ls.interest_amount <> if(ls.accrual_interest_derived is null,0,ls.accrual_interest_derived)))")
-                .append("  and loan.loan_status_id=? and mpl.accounting_type=? and loan.is_npa=0 and ls.duedate <= CURDATE() order by loan.id,ls.duedate");
+                .append("  and (loan.loan_status_id=? or loan.expected_maturedon_date >= CURDATE() or loan.expected_maturedon_date > IFNULL(loan.accrued_till, '1900-01-01')) and mpl.accounting_type=? and loan.is_npa=0 and ls.duedate <= CURDATE() order by loan.id,ls.duedate");
```

Comment:
Hi Pramod, this depends on the region you are working in. You cannot just accrue all remaining income in go, it goes against accounting principles. Therefore the batch job (for flat loans) should just recognise in line with schedule as before. The main issue was that no bookings were made at all before, which left a loan with missing income accruals which is clearly wrong as well.